lib/rules/prefer-read-only-props.js

```diff
@@ -50,38 +57,62 @@ module.exports = {
     schema: [],
   },
 
-  create: Components.detect((context, components) => ({
-    'Program:exit'() {
-      flatMap(
-        values(components.list()),
-        (component) => component.declaredPropTypes || []
-      ).forEach((declaredPropTypes) => {
-        Object.keys(declaredPropTypes).forEach((propName) => {
-          const prop = declaredPropTypes[propName];
-
-          if (!prop.node || !isFlowPropertyType(prop.node)) {
-            return;
-          }
-
-          if (!isCovariant(prop.node)) {
-            report(context, messages.readOnlyProp, 'readOnlyProp', {
-              node: prop.node,
-              data: {
-                name: propName,
-              },
-              fix: (fixer) => {
-                if (!prop.node.variance) {
-                  // Insert covariance
-                  return fixer.insertTextBefore(prop.node, '+');
+  create: Components.detect((context, components) => {
+    function reportReadOnlyProp(prop, propName, fixer) {
+      report(context, messages.readOnlyProp, 'readOnlyProp', {
+        node: prop.node,
+        data: {
+          name: propName,
+        },
+        fix: fixer,
+      });
+    }
+
+    return {
+      'Program:exit'() {
+        flatMap(
+          values(components.list()),
+          (component) => component.declaredPropTypes || []
+        ).forEach((declaredPropTypes) => {
+          Object.keys(declaredPropTypes).forEach((propName) => {
+            const prop = declaredPropTypes[propName];
+            if (!prop.node) {
+              return;
+            }
+
+            switch (prop.node.type) {
```

Comment:
let's avoid switch please; an if/else is nicer.

Reply:
Done, the failing tests are related to `semver` dependency requiring a major upgrade

Reply:
they're a bit more involved than that; we can't upgrade semver (and it's a false positive non-vulnerability anyways), it's that the tests are failing on master.